src/toxcore/tcp/server/server_ext.rs

```diff
@@ -1,28 +1,71 @@
-/*! Extension trait for run TCP server on `TcpStream`
+/*! Extension trait for run TCP server on `TcpStream` and ping sender
 */
 
 use std::io::{Error, ErrorKind};
 use std::time::{Duration, Instant};
 
 use futures::{future, Future, Sink, Stream};
-use tokio::net::TcpStream;
+use tokio::net::{TcpStream, TcpListener};
 use tokio::util::FutureExt;
 use tokio_codec::Framed;
+use tokio::timer::Interval;
 
 use toxcore::crypto_core::*;
 use toxcore::io_tokio::IoFuture;
 use toxcore::tcp::codec::Codec;
 use toxcore::tcp::handshake::make_server_handshake;
 use toxcore::tcp::server::{Server, ServerProcessor};
 
-/// Extension trait for running TCP server on incoming `TcpStream`
+/// Interval in seconds for Tcp Ping sender
+const TCP_PING_INTERVAL: u64 = 1;
+
+/// Extension trait for running TCP server on incoming `TcpStream` and ping sender
 pub trait ServerExt {
+    /// Running TCP ping sender and incoming `TcpStream`
+    fn run(self: Self, listner: TcpListener, dht_sk: SecretKey) -> IoFuture<()>;
     /// Running TCP server on incoming `TcpStream`
-    fn run(self: Self, stream: TcpStream, dht_sk: SecretKey) -> IoFuture<()>;
+    fn run_connection(self: Self, stream: TcpStream, dht_sk: SecretKey) -> IoFuture<()>;
 }
 
 impl ServerExt for Server {
-    fn run(self: Self, stream: TcpStream, dht_sk: SecretKey) -> IoFuture<()> {
+    fn run(self: Self, listner: TcpListener, dht_sk: SecretKey) -> IoFuture<()> {
+        let self_c = self.clone();
+
+        let future = listner.incoming()
+            .for_each(move |stream|
+                self.clone().run_connection(stream, dht_sk.clone())
+            )
+            .map_err(|err| {
+                // All tasks must have an `Error` type of `()`. This forces error
+                // handling and helps avoid silencing failures.
+                //
+                // In our example, we are only going to log the error to STDOUT.
```

Comment:
Outdated comment.
Btw, do we need this log at all? It can be logged outside in `tox-node`.